src/components/main-form.old.js

```diff
@@ -0,0 +1,184 @@
+            <div className="col-6">
+              <h4>Mortgage 2</h4>
+              <label>What is the mortgage's APR? (annual percentage rate)</label>
+              <Input
+                className="mb-2"
+                name="options.b.mortgageInterestRate"
+                onChange={this.updateInput}
+                placeholder="Annual Percentage Rate"
+                type="number"
+                value={b.mortgageInterestRate}
+              />
+              <label>What is the term of the second mortgage? (in years)</label>
+              <Input
+                className="mb-2"
+                name="options.b.mortgageTerm"
+                onChange={this.updateInput}
+                placeholder="Number of years"
+                type="number"
+                value={b.mortgageTerm}
+              />
+            </div>
+          </div>
+          <label>What is the size of your loan?</label>
+          <Input
+            className="mb-2"
+            name="mortgageAmt"
+            onChange={this.updateInput}
+            placeholder="Mortgage amount"
+            type="number"
+            value={this.state.mortgageAmt}
+          />
+          <label>What is your expected return if you invest your money?</label>
+          <Input
+            className="mb-2"
+            name="investmentRate"
+            onChange={this.updateInput}
+            placeholder="Expected return for your investments"
+            type="number"
+            value={this.state.investmentRate}
+          />
+        </div>
+        <div className="col-8">
+          {this.validateData(this.state) ?
+            <div>
+              <p>
+                This is the amount of money you'll have if you invest ${numeral(-1 * (b.pmt - a.pmt).toFixed(0)).format('0,0')} on a monthly basis at a {numeral(this.state.investmentRate).format('0,0')}% annual return rate after {numeral(a.mortgageTerm).format('0,0')} years, minus a total mortgage interest of ${numeral(-1 * a.interestAmt.toFixed(0)).format('0,0')}: <b>${numeral(gainForA.toFixed(0)).format('0,0')}</b>
+              </p>
+              <p>
+                This is the amount of money you'll have if you invest ${numeral(-1 * b.pmt.toFixed(0)).format('0,0')} on a monthly basis at a {numeral(this.state.investmentRate).format('0,0')}% annual return rate after {numeral(a.mortgageTerm - b.mortgageTerm).format('0,0')} years, minus a total mortgage interest of ${numeral(-1 * b.interestAmt.toFixed(0)).format('0,0')}: <b>${numeral(gainForB.toFixed(0)).format('0,0')}</b>
+              </p>
+              <p>
+                Opportunity cost (the amount of money gained/lost by going with a {numeral(a.mortgageTerm).format('0,0')} year mortgage): <b>${numeral((gainForA - gainForB).toFixed(0)).format('0,0')}</b>
+              </p>
+            </div>
+            : <p>Please fill out all the inputs/fields on the left.</p>
+          }
+        </div>
+      </div>
+    )
+  }
+}
+
+export default MainForm;
```

Comment:
This can all be deleted thanks to version control

Reply:
definitely, just left it there in case we needed to copy paste any more code without relying on version control

Reply:
That's easy to do w/ a prior version of Github

Reply:
even easier to just copy paste. It's just a matter of comfort. That way I don't have to go into Github or use git to read over old code in order to see how we used to do things. We can delete it when we're done with the second version. It does no harm imo.

Reply:
yes, though the trade-off is having more non-running code around

Reply:
https://github.com/jgozal/moneycompass/blob/14835251211f027f78e170b4ffe996e30325a19a/src/components/main-form.js

Reply:
The project is literally 2 scripts. I don't really see the trade-off. Like I said, it's easier for me to just switch my tab in my editor when I need to review old code like this, and I don't see how this could disturb your workflow. It's completely fine to have old pieces of code hanging around in dev as long as it's clear. I think this is a non-issue and it doesn't deserve 7 comments lol.

Reply:
The project is literally 2 scripts. I don't really see the trade-off. Like I said, it's easier for me to just switch my tab in my editor when I need to review old code like this. I had to do this action perhaps 20-30 times when I was refactoring everything, so it's just a lot easier to not have to depend on an external source to do that. I also don't see how this could disturb your workflow, perhaps I'm missing something... It's completely fine to have old pieces of code hanging around in dev as long as it's clear. I think this is a non-issue and it doesn't deserve 7 comments lol. We both know this will never get to prod.

Reply:
In my experience I've lost time investigating bugs in code that actually never gets run. The mental load is less for the writer, and more for the readers coming back in a few weeks.

90% of a developer's time is spent reading code, so it's a good rule of thumb to use the existing version control features that support this workflow.

You are absolutely right that in this particular situation, the trade-off is minimal. You may want to know the reasoning for future code bases.

Reply:
I try to be situational about my decisions and don't really like rules of thumb as you know.  In this particular case it doesn't really matter. 

I obviously see how this could be a problem if I were leaving small pieces of **undocumented** old code in the "active" script however, and I would never do that. I write **undocumented** because I think it's also fine to leave small pieces of non-working, **documented** old code in an active script **in dev**. It's pretty common actually and I would be surprised if you've never done it yourself or worked with developers that do it. It serves as a reminder sometimes to do x, y, and z in a different way from the one previously implemented. If you remove it, the reminder is no longer there in front of you and you risk forgetting about it. As long as the developer is explicit about that being old, non-working code, it should be harmless. This is no different from what I did here and the _.old_ file name ending is explicit enough imo.